src/Protocol/ActivityPub/Processor.php

```diff
@@ -64,10 +64,9 @@ private static function replaceEmojis($body, array $emojis)
 	 *
 	 * @param array   $tags
 	 * @param boolean $sensitive
-	 * @param array   $implicit_mentions List of profile URLs to skip
 	 * @return string with tags
 	 */
-	private static function constructTagString(array $tags, $sensitive)
+	private static function constructTagString(array $tags = null, $sensitive = false)
```

Comment:
```suggestion
	private static function constructTagString(array $tags = [], $sensitive = false)
```
setting an array to null doesn't seems like an good idea

Reply:
If you look at the Fatal Error in https://github.com/friendica/friendica/issues/6916#issuecomment-493218564, you'll notice that it happens because a `null` value is given for `$tags` and it triggers PHP's internal type check since we provided a type-hint in the method declaration.

I initially tried going up the call chain but I couldn't find a way to ensure `$activity['tags']` was set. There are two other ways of mitigating such error by modifying the declaration.

Either the PHP < 7 version:
```php
private static function constructTagString(array $tags = null, $sensitive = false)
```
or the PHP >= 7 version that your comment actually reminded me:
```php
private static function constructTagString(array? $tags, $sensitive)
```

Since we now support PHP version 7 minimum, I can actually go for the second form which would be more appropriate than the first.

This way we don't have to use the inelegant `defaults()` in the method call and the `empty()` test can still catch a `null` value.